### PR TITLE
fix(preview): use new autodocs configuration format

### PR DESCRIPTION
**Pull request by jawinn**
## Description

Updates the way that autodocs is configured, as documented in the [Storybook migration guide
for 8.0.x to 8.1.x](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#mainjs-docsautodocs-is-deprecated). The `docs.autodocs` setting in `main.js` was deprecated and is being removed in Storybook 9.0.

For the stories using custom MDX files, it was necessary to add `!autodocs` to their default story. Otherwise Storybook throws a bunch of errors like the following and would not load:
> "You created a component docs page for 'Components/*', but also tagged the CSF file with 'autodocs'. This is probably a mistake."

Includes an update to the "Contributing" guide in Storybook to clarify how tags are used on stories.

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

- [x] The "Docs" pages are still appearing for various components. Including those with autodocs and those with custom MDX files. @marissahuysentruyt (components with MDX files that were checked: actionbar, accordion, asset, assetcard @rise-erpelding : badge, calendar, card, coachmark, coloarea, colorhandle, colorloupe, colorslider, colorwheel, combobox, contextualhelp, datepicker, fieldgroup (which is broken anyway but not because of this work), and icon)
- [x] Stories with `!autodocs` continue to not appear (for example, With Forced Colors stories). @marissahuysentruyt (checked buttongroup, tabs, switch, picker) @rise-erpelding checked actionbutton, asset, badge, button, clearbutton, datepicker, icon, popover, splitview
- [x] Proofread tags section of Storybook "Contributing" docs

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [x] I have updated relevant storybook stories and templates.
- [x] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: .storybook/guides/develop.mdx
Comment (jawinn) [inline .storybook/guides/develop.mdx:448]: I removed the mentions of the `test` tag here, as the Storybook docs say the test tag is for Storybook's test runner:
>Stories tagged with test do not currently affect Storybook's UI, but can be used to filter the [test runner](https://storybook.js.org/docs/writing-tests/test-runner#run-tests-for-a-subset-of-stories).

The `test` tag is also already applied by default to all stories in Storybook.

Comment (castastrophe) [reply .storybook/guides/develop.mdx:448]: Do we want to include removing the "test" tag in this PR from stories where they are using them now?

File: components/accordion/stories/accordion.stories.js
Comment (castastrophe) [inline components/accordion/stories/accordion.stories.js:69]: Since the `!autodocs` is being set at the story's default, do we need to keep the autodocs tag in the Compact and Spacious stories to turn them back on?

Comment (castastrophe) [reply components/accordion/stories/accordion.stories.js:69]: (Ditto for a few other stories in this list with the same update)

Comment (jawinn) [reply components/accordion/stories/accordion.stories.js:69]: The ones with default `!autodocs` added like this use a custom MDX template that defines what stories are displayed (accordion already shows compact and spacious). I'm not sure if Storybook will fix it in the future, but right now it causes the error noted in the PR description if the components using a custom MDX were tagged with the default `autodocs`.

File: .storybook/guides/develop.mdx
Comment (jawinn) [reply .storybook/guides/develop.mdx:448]: Sure. I've removed them in an added commit, [7df05a1](https://github.com/adobe/spectrum-css/pull/2924/commits/7df05a19867ba66df2ae7751636b0946543acb79)

File: components/underlay/stories/underlay.stories.js
Comment (marissahuysentruyt) [inline components/underlay/stories/underlay.stories.js:36]: Hm, why did we have `!test` for this default? Wouldn't that remove it from Chromatic? I see an [underlay snapshot from main](https://www.chromatic.com/component?appId=64762974a45b8bc5ca1705a2&csfId=components-underlay&buildNumber=2477&k=66a27c9ae1a436a8461a65f7-Context%3A+Express-interactive-true&h=48&b=-1) though (I think) from a while back. Just curious 😊

File: components/pickerbutton/stories/pickerbutton.stories.js
Comment (marissahuysentruyt) [inline components/pickerbutton/stories/pickerbutton.stories.js:131]: This is unrelated to the purpose of this PR, but why are the Quiet and Disabled stories removed the docs page? Just wondering.

**Review by marissahuysentruyt — COMMENTED** [review]
Great idea setting us up for the future and Storybook 9.

This is all looking good, I only had a few small things to note. Mainly questions and a so-so wording change. Nice work!

**Review by rise-erpelding — COMMENTED** [review]
This is looking great! I put in my two cents on the `develop.mdx` and I did look at many components and Docs pages as noted in the PR description, and everything appears to be in order there. VRTs look clear as well, the only changes are the same ones that I caused last week! 😬

File: components/underlay/stories/underlay.stories.js
Comment (jawinn) [reply components/underlay/stories/underlay.stories.js:36]: It wasn't doing anything previously. I'm not sure why it was there.

**Review by marissahuysentruyt — APPROVED** [review]
This is looking good to me. I'm approving now, but if you need to me review once we've gotten some other PRs merged, just lemme know!

**Review by rise-erpelding — APPROVED** [review]
This looks great, thank you Josh!

**Review by pfulton — APPROVED** [review]

**Pull request merged**
